### PR TITLE
Create Vector_right_rotation_linear_time.cpp

### PR DESCRIPTION
**Pull request by vinay6023**
This is one of the fastest implementation among vector/ array rotating algorithms .

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1351"></a>

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1351"></a>

### CODE REVIEW

**Comment by kvedala** [comment]
@vinay6023 

Is this a duplicate of an already open pull-request #1288 ?

There are 5 duplicate pull-requests for left rotation and 7 duplicate pull-requests for right rotation.
Left rotation: #1279, #1281, #1282 #1284 
Right rotation: #1285 , #1288 , #1347 #1348 #1349 #1350 and #1351 

You do not have to create multiple pull-requests. This is otherwise mark your account as a spammer.

After creating a pull request, you can add and push new commits to the branch and they will show up in the pull-request as well.

If you want to contribute properly, read the guidelines and follow the checklist. If you are simply dumping your code without following any of the readme or the provided checklist, your account can be marked as a spammer and blocked.

File: operations_on_datastructures/Vector_right_rotation_linear_time.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include<vector>
+/*
+Lets place place every number of the array at its desired correct position. But while doing this, we will lose the original elements.
```
Comment (vinay6023) [inline operations_on_datastructures/Vector_right_rotation_linear_time.cpp:4]: explanation

**Comment by vinay6023** [comment]
@kvedala 
Sorry, for the inconvenience. I don't know how this happened ,as i had closed all the pull requests except one. Also could you help me with the automated testing, as i had submitted the similar code for an problem on codeforces and it was accepted ,I am having a difficult time figuring out what are the exceptions this code is missing.

**Comment by kvedala** [comment]
> @kvedala 
> Sorry, for the inconvenience. I don't know how this happened ,as i had closed all the pull requests except one. Also could you help me with the automated testing, as i had submitted the similar code for an problem on codeforces and it was accepted ,I am having a difficult time figuring out what are the exceptions this code is missing.
> 

Step# 1
Follow the checklist and the links should answer all your questions. Once the code is updated to the repo standards, then the reviewers will review and can help you.
Until the code is updated to the guidelines, no further steps will be taken by the reviewers.

**Review by vinay6023 — COMMENTED** [review]
@kvedala 
I have tried to implement the changes you asked for to the best of my understanding,
i request you to kindly  review them and suggest improvements

**Comment by vinay6023** [comment]
-documentation
-test cases
-

**Comment by vinay6023** [comment]
@Panquesito7 could you please throw some light over what other modifications are needed to make it an repo standard code.

**Pull request closed without merging**